### ISSUE TITLE
2.0.0 Waldorf Blofeld adaptation does not detect Blofeld properly

### ISSUE THREAD

**Issue by bboc** [christofmuc/KnobKraft-orm#218]
Waldorf Blofeld is detected, the bar is green, but the channel detected is 1, while the channel selected in the Blofeld is in fact 12. 

Apparently, Device ID is assumed to be "0", which should be mentioned in the help text!

When I click on a bank, patches are imported from Blofeld, but patch names don't show up in KnobCraft. 

Now when I change the MIDI channel in KnobCraft to the actual channel (12), KnobKraft sends program changes to Blofeld, but receiving banks no longer works.

Friendly patch names and friendly bank names are not working, either.

**Comment by christofmuc** [comment]
@bboc Hm, this is strange, I though I had feedback before that Blofeld was working. Can you attach some data here, like a non-working patch? 

It could be we suffer from my original confusion of device ID and MIDI Channel, and the fact that the DSI/Sequentials work completely with one or the other, and I had originally developed for those.

Question - the Blofeld has a MIDI channel, but that is the same for sending and receiving the data, right?
Device ID might be different, but we need to check if it is detected appropriately.

**Comment by christofmuc** [comment]
Ok, confirmed - the old Blofeld adaptation expects channel = device ID. This is a bug that can be fixed, if we find out how to coax the channel from the device.

If we look at 
https://www.deepsonic.ch/deep/docs_manuals/waldorf_blofeld_sysex_documentation_v.1.04.txt

we can see we can harvest the MIDI channel from index 35 of the global parameter request.

**Comment by bboc** [comment]
That still doesn't solve the patch names, they are shown as "Bank-Program":

<img width="696" alt="image" src="https://user-images.githubusercontent.com/520442/219138103-0306da03-a600-456b-9343-bb3f10c045ef.png">

**Comment by bboc** [comment]
Ok, now I see: the Blofeld adaptation doesn't contain any code for extracting names, renaming, fingerprinting etc. Do you have a Blofeld? Otherwise I can take the spec in the link above and see if I can fix it, once my current adaptation for the Modor NF-1m is finished.

**Comment by christofmuc** [comment]
That would be great! No, I don't have a Blofeld, so maybe this was just a proof of concept to see if the adaptation interface would work. The adaptation view in the KnobKraft is a quick way to see the capabilities of the respective adaptation.

Actually the Blofeld Adaptation has nameFomDump() in line 96ff, so it should work, maybe it is suffering from the other import problem. Can you try to load a sysex file from disk and see if names appear there? Then the problem is the same as issue #217.

**Comment by bboc** [comment]
I'll fix the AS-1 first, and then I'll look at the Blofeld. The sysex implementation seems pretty straightforward. I'll implement all the missing functions as well, maybe with the exception of Bank Dump, which appears unnecessary. Or ist that capability also used for importing sysex files?

**Comment by christofmuc** [comment]
Yes, it would be used if you would encounter a bank dump sysex file. But I'd wait until that actually happens, I don't think they are very common (and an import will log warnings for sysex messages that are not understood).

**Comment by bboc** [comment]
I just tried importing a bank from a sysex file, the names are also not decoded properly there. It looks like nameFromDump() is not even called. 

However, I noted that after I trigger the file import, the log literally says 1000 times that the built-in automatic category rules are overridden, then it tells me another thousand times that everything is a duplicate, even though there is no function defined for calculating a fingerprint. I have just used a random dump from my computer that I downloaded somewhere, will have to test with one that I created myself.

**Comment by christofmuc** [comment]
Ah, I have a note somewhere about this. The old log system had a call that would actually log a specific message only once per user session, I think this is one of them. I'll open a new issue to get this fixed, very annoying.

**Comment by christofmuc** [comment]
@bboc I made a new bug for the duplicate log lines, and the name not being extracted must be the same as #217, because the Waldorf Blofeld adaptation does not implement renamePatch(), and due to the bug this prevent nameFromDump() being called. I'll fix that with the next release.

If you have an updated adaptation for the Blofeld with the global dump based detection, I can include that in the release. Feel free to just attach the file here, no need to a pull request.

**Comment by bboc** [comment]
I'm not finished with the Blofeld yet.

**Comment by christofmuc** [comment]
No problem

**Comment by christofmuc** [comment]
Closing as stale - let me know if you have a newer version of the Blofeld adaptation!
